MIG-1453: updating upgrade documentation (#65453)
openshift/openshift-docs: modules/upgrading-mtc-1-8-0.adoc

// Module included in the following assemblies:
//
// * migration_toolkit_for_containers/upgrading-mtc.adoc

:_content-type: PROCEDURE
[id="migration-upgrading-mtc-18_{context}"]
= Upgrading to {mtc-full} to 1.8.0

.Procedure

> * Determine subscription name(s) and current channels to work with for upgrading, by using either issuing:

Reviewer: ```suggestion
* Determine subscription names and current channels to work with for upgrading, by either issuing:
```

"_... upgrading, by using either issuing:_" doesn't read well to me.

+
[source,terminal]
----
$ oc -n openshift-migration get sub
----
+
Which will return an output similar to the following:

+
[source,terminal]
----
NAME                                                                         PACKAGE                SOURCE                 CHANNEL
mtc-operator                                                                 mtc-operator           mtc-operator-catalog   release-v1.7
redhat-oadp-operator-stable-1.0-mtc-operator-catalog-openshift-marketplace   redhat-oadp-operator   mtc-operator-catalog   stable-1.0
----
+

> Or for a more programmatic approach, issue:

Reviewer: ```suggestion
Or, for a more programmatic approach, issuing:
```
Shouldn't both uses of 'issue' be consistent? Use 'issuing' here as well?

+
[source,terminal]
----
$ oc -n openshift-migration get sub -ojson | jq -r '.items[] | { name: .metadata.name, package: .spec.name, channel: .spec.channel }'
----
+
Which will return an output similar to the following:
+
[source,terminal]
----
{
  "name": "mtc-operator",
  "package": "mtc-operator",
  "channel": "release-v1.7"
}
{
  "name": "redhat-oadp-operator-stable-1.0-mtc-operator-catalog-openshift-marketplace",
  "package": "redhat-oadp-operator",
  "channel": "stable-1.0"
}
----

* For each subscription, patch to move from the {mtc-short} 1.7 channel to the newer channel {mtc-short} 1.8:
[source,terminal]
----
$ oc -n openshift-migration patch subscription mtc-operator --type merge --patch '{"spec": {"channel": "release-v1.8"}}'
----
+

> The subscription is returned as `patched`:

Reviewer: I'm not sure what is required here to fix it, but the formatting in the rendered preview has a plus `+` in front of "_The subscription..._" text rather than indenting the line to align with the rest of the block under the bullet.

Author: fixed - will check build preview

// Module included in the following assemblies:
//
// * migration_toolkit_for_containers/upgrading-mtc.adoc

:_content-type: PROCEDURE
[id="migration-upgrading-mtc-18_{context}"]
= Upgrading to {mtc-full} to 1.8.0

.Procedure

* Determine subscription names and current channels to work with for upgrading, by either issuing:

+
[source,terminal]
----
$ oc -n openshift-migration get sub
----
+
Which will return an output similar to the following:

+
[source,terminal]
----
NAME                                                                         PACKAGE                SOURCE                 CHANNEL
mtc-operator                                                                 mtc-operator           mtc-operator-catalog   release-v1.7
redhat-oadp-operator-stable-1.0-mtc-operator-catalog-openshift-marketplace   redhat-oadp-operator   mtc-operator-catalog   stable-1.0
----
+
Or, for a more programmatic approach, issuing:
+
[source,terminal]
----

> $ oc -n openshift-migration get sub -ojson | jq -r '.items[] | { name: .metadata.name, package: .spec.name, channel: .spec.channel }'

Reviewer: ```suggestion
$ oc -n openshift-migration get sub -o json | jq -r '.items[] | { name: .metadata.name, package: .spec.name, channel: .spec.channel }'
```

----
+
Which will return an output similar to the following:
+
[source,terminal]
----
{
  "name": "mtc-operator",
  "package": "mtc-operator",
  "channel": "release-v1.7"
}
{
  "name": "redhat-oadp-operator-stable-1.0-mtc-operator-catalog-openshift-marketplace",
  "package": "redhat-oadp-operator",
  "channel": "stable-1.0"
}
----

* For each subscription, patch to move from the {mtc-short} 1.7 channel to the newer channel {mtc-short} 1.8:
+
[source,terminal]
----
$ oc -n openshift-migration patch subscription mtc-operator --type merge --patch '{"spec": {"channel": "release-v1.8"}}'
----
+
The subscription is returned as `patched`:
+
[source,terminal]
----
subscription.operators.coreos.com/mtc-operator patched

> ----

Reviewer: This block is out-dented from the rest of the bullet content preceding it when looking at the preview.

Author: I think it is ok in the preview:

![image](https://github.com/openshift/openshift-docs/assets/106804821/2438444e-c213-42d5-ae5e-83103cea8919)

Reviewer: This too now shows up as expected for me when I preview today.

////
moving to a new oadp 1

OADP 1.0 to 1.2:

$ oc -n openshift-migration patch subscription redhat-oadp-operator-stable-1.0-mtc-operator-catalog-openshift-marketplace --type merge --patch '{"spec": {"channel":"stable-1.2"}}'

subscription.operators.coreos.com/redhat-oadp-operator-stable-1.0-mtc-operator-catalog-openshift-marketplace patched

NOTE: If installPlanApproval is "Automatic", OLM will begin the upgrade process.  If it is "Manual", the user will need to approve each installPlan before OLM will begin the upgrades.

Wait for OLM to complete the upgrades of OADP and MTC.

> 
>     ...
> 

Reviewer: Remove extraneous/blank lines
```suggestion
```

> $ oc -n openshift-migration get subscriptions.operators.coreos.com mtc-operator  -o json | jq '.status | (."state"=="AtLatestKnown")'

Reviewer: ```suggestion
$ oc -n openshift-migration get subscriptions.operators.coreos.com mtc-operator -o json | jq '.status | (."state"=="AtLatestKnown")'
```

true

Verify the channel used for each subscription.

> $ oc -n openshift-migration get sub -ojson | jq -r '.items[] | {name: .metadata.name, channel: .spec.channel }'

Reviewer: ```suggestion
$ oc -n openshift-migration get sub -o json | jq -r '.items[] | {name: .metadata.name, channel: .spec.channel }'
```

// Module included in the following assemblies:
//
// * migration_toolkit_for_containers/upgrading-mtc.adoc

:_content-type: PROCEDURE
[id="migration-upgrading-mtc-18_{context}"]
= Upgrading to {mtc-full} to 1.8.0

To upgrade to {mtc-full} to 1.8.0, complete the following steps.

.Procedure

. Determine subscription names and current channels to work with for upgrading by using one of the following methods:

> ** <Description of what this command does, versus the other one> by running the following command:

Author: @bscott-rh - updated. thanks
